src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -308,10 +312,15 @@ public static void PreloadAsmFromPath(string preloaderLocation, string asmLocati
                 throw new MissingMethodException(
                     string.Format("Method '{0}' not found", PreloaderMethodName));
             }
-
-            var methodParams = new object[] { asmLocation };
-            preloadMethod.Invoke(null, methodParams);
-
+            try
+            {
+                var methodParams = new object[] { asmLocation };
+                preloadMethod.Invoke(null, methodParams);
+            }
+            catch
+            {
+                throw new FileNotFoundException($"Could not load geometry library binaries from : {asmLocation}");
```

Comment:
Is this the only kind of cause exception? @mjkkirschner @aparajit-pratap

Reply:
this should just be `Exception` - I will change it.